src/components/chat-components/RelevantNotes.tsx

```diff
@@ -0,0 +1,261 @@
+import { Badge } from "@/components/ui/badge";
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@/components/ui/collapsible";
+import { findRelevantNotes, RelevantNoteEntry } from "@/search/findRelevantNotes";
+import VectorStoreManager from "@/search/vectorStoreManager";
+import { Tooltip, TooltipContent, TooltipTrigger } from "@/components/ui/tooltip";
+import { ArrowRight, ChevronDown, ChevronUp, PlusCircle, TriangleAlert } from "lucide-react";
+import React, { forwardRef, useEffect, useState } from "react";
+import { TFile } from "obsidian";
+import { Popover, PopoverContent, PopoverTrigger } from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+
+function useRelevantNotes() {
+  const [relevantNotes, setRelevantNotes] = useState<RelevantNoteEntry[]>([]);
+  const activeFile = app.workspace.getActiveFile();
+
+  useEffect(() => {
+    async function fetchNotes() {
+      if (!activeFile?.path) return;
+      const db = await VectorStoreManager.getInstance().getDb();
+      const notes = await findRelevantNotes({ db, filePath: activeFile.path });
+      setRelevantNotes(notes);
+    }
+    fetchNotes();
+  }, [activeFile?.path]);
+
+  return relevantNotes;
+}
+
+function useHasIndex(notePath: string) {
+  const [hasIndex, setHasIndex] = useState(true);
+  useEffect(() => {
+    if (!notePath) return;
+    async function fetchHasIndex() {
+      const hasIndex = await VectorStoreManager.getInstance().hasIndex(notePath);
+      setHasIndex(hasIndex);
+    }
+    fetchHasIndex();
+  }, [notePath]);
+  return hasIndex;
+}
+
+const ExplainerBadge = forwardRef<HTMLDivElement, { children: React.ReactNode }>(
+  ({ children, ...props }, ref) => {
+    return (
+      <div ref={ref} {...props} className="flex gap-2">
+        <Badge variant="secondary" className="text-xs text-muted whitespace-nowrap cursor-default">
+          {children}
+        </Badge>
+      </div>
+    );
+  }
+);
+
+function inSameFolder(path1: string, path2: string) {
+  const folder1 = path1.split("/").slice(0, -1).join("/");
+  const folder2 = path2.split("/").slice(0, -1).join("/");
+  return folder1 === folder2;
+}
+
+function SimilarityBadge({ score }: { score: number }) {
+  let text = "🔴 Low Similarity";
+  if (score > 0.5) text = "🟠 Medium Similarity";
+  if (score > 0.7) text = "🟢 High Similarity";
```

Comment:
From my quick test with copilot-plus-small, ~80% of the similar notes fall in Medium Similarity, ~10+% High Similarity and rarely Low Similarity. Of course this distribution depends heavily on the embedding model used. Not really a problem but just to report an observation.